src/System.Data.SqlClient/src/Microsoft/SqlServer/Server/MetadataUtilsSmi.cs

```diff
@@ -77,57 +73,60 @@ internal class MetaDataUtilsSmi
             SqlDbType.DateTimeOffset,       // System.DateTimeOffset
         };
 
-        // Hash table to map from clr type object to ExtendedClrTypeCodeMap enum
-        // this HashTable should only be accessed from DetermineExtendedTypeCode and class ctor for setup.
-        private static readonly Hashtable s_typeToExtendedTypeCodeMap = CreateTypeToExtendedTypeCodeMap();
+        // Dictionary to map from clr type object to ExtendedClrTypeCodeMap enum.
+        // This dictionary should only be accessed from DetermineExtendedTypeCode and class ctor for setup.
+        private static readonly Dictionary<Type, ExtendedClrTypeCode> s_typeToExtendedTypeCodeMap = CreateTypeToExtendedTypeCodeMap();
```

Comment:
@stephentoub Was just a suggestion :smile: I realize I've been submitting a lot of PRs to eliminate static data structure allocations lately, and that readable code is preferable over one that uses a bunch of ifs/cases, but there are some places that do need it (perhaps not here though). Will try to limit my PRs to places where it can easily be replaced.